kyaml/yaml/fns.go

```diff
@@ -165,6 +180,10 @@ func MatchElement(field, value string) ElementMatcher {
 	return ElementMatcher{FieldName: field, FieldValue: value}
 }
 
+func GetElementByKey(key string) ElementMatcher {
+	return ElementMatcher{FieldName: key, NoValue: true}
+}
+
 // ElementMatcher returns the first element from a Sequence matching the
 // specified field's value.
```

Comment:
// If there's no match, and no configuration error, the matcher returns nil, nil.